AlphaWallet/Core/Helpers/BlockieGenerator.swift

```diff
@@ -85,24 +91,27 @@ class BlockiesGenerator {
         let promise: Promise<String>
 
         if let ens = ens {
-            promise = GetENSTextRecord(server: .forResolvingEns)
-                .getENSRecord(forName: ens, record: .avatar)
+            promise = ensTextRecordFetcher.getENSRecord(forName: ens, record: .avatar)
         } else {
-            promise = GetENSTextRecord(server: .forResolvingEns)
-                .getENSRecord(forAddress: address, record: .avatar)
+            promise = ensTextRecordFetcher.getENSRecord(forAddress: address, record: .avatar)
         }
 
         return firstly {
             promise
         }.then { url -> Promise<BlockiesImage> in
```

Comment:
on next call the url will be here, so we don't need to fetch image from open sea again

Reply:
` }.then { url -> Promise<BlockiesImage> in`

Reply:
Is this responding to my comment at https://github.com/AlphaWallet/alpha-wallet-ios/pull/4805#discussion_r894139983?

I mean `fetchEnsAvatar()` has this to cache the result, but is that cached value ever read from the cache again?

```
let key = EnsLookupKey(nameOrAddress: ens ?? address.eip55String, server: .forResolvingEns, record: .avatar)
self.storage.addOrUpdate(record: .init(key: key, value: .record(url.absoluteString)))
```

Reply:
i.e.

```
if isCached
    use_cached //I can't find this
else
    fetch
    cache //it does this
end
```

Reply:
we update it as ens record `avatar`, so it going to be returned in GetEnsTextRecord